src/base/async/int.c

```diff
@@ -1834,10 +1839,10 @@ static int int19(void)
 uint16_t RedirectDevice(char *dStr, char *sStr,
                         uint8_t deviceType, uint16_t deviceParameter)
 {
+  // may be nested, so use local var unstead of pre/post_msdos()
+  struct vm86_regs local_regs = REGS;
```

Comment:
But do you know where exactly did
the second save happened? Maybe it
would be possible to un-recurse?
Also we can maintain the stack of
the saved registers if  this is really needed.
So it would be nice to know when exactly
this happens.